src/search.js

```diff
@@ -1,31 +1,31 @@
-import {assoc, concat, filter, has, isEmpty, map, pipe, prop, sortBy} from "ramda"
+import {ascend, assoc, concat, descend, has, isEmpty, map, prop, sortBy, sortWith} from "ramda"
 import * as diacritics from 'diacritics'
 
 
 export function findParametersAndVariables(parameters, variables, query) {
-  function matchesQuery(parameterOrVariable) {
-    if (isEmpty(normalizedQuery)) {
-      return true
-    }
-    return parameterOrVariable.name.includes(normalizedQuery) ||
-      parameterOrVariable.searchIndex.includes(normalizedQuery)
+  function weightByPositionIn(string, substring) {
+    const index = string.indexOf(substring)
+    return index === -1 ? Number.MAX_SAFE_INTEGER : index
   }
   const normalizedQuery = diacritics.remove(query.toLowerCase())
-  return pipe(
-    filter(matchesQuery),
-    sortBy(prop('name')),
-  )(concat(
-    map(assoc('type', 'parameter'), parameters),
-    map(assoc('type', 'variable'), variables),
-  ))
+  const parametersAndVariables = concat(parameters, variables)
+  if (isEmpty(normalizedQuery)) {
+    return sortBy(prop('name'), parametersAndVariables)
+  } else {
+    return sortWith([
+      ascend(item => weightByPositionIn(item.name, normalizedQuery)),
+      descend(item => item.normalizedDescription.includes(normalizedQuery)),
+      ascend(prop('name')),
+    ], parametersAndVariables)
+  }
 }
 
-export function withSearchIndex(propertyName, objects) {
+export function withNormalizedDescription(objects) {
```

Comment:
I agree, but in that case the input data is not mutated. In a language which allows both mutating and non-mutating functions, I like to use:
- `addFoo(obj)` when doing `obj.foo = "bar"`
- `withFoo` when doing `Object.extend({}, foo, "bar")` (or `assocFoo` to follow RamdaJS naming, but can be considered odd)

> One of the more universal, yet simple rules is: Function names should be verbs if the function changes the state of the program, and nouns if they're used to return a certain value.

Besides, I think in that case that this function matches the second part of the sentence, but that's a modified copy of the input, too :)

So OK for `addNormalizedDescription` and keep in mind it's not a mutating function!

Sorry for this long answer for a such small point :-D